DXMainClient/DXGUI/Multiplayer/CnCNet/PasswordRequestWindow.cs

```diff
@@ -74,7 +78,16 @@ private void Keyboard_OnKeyPressed(object sender, KeyPressEventArgs e)
         private void PasswordRequestWindow_EnabledChanged(object sender, EventArgs e)
         {
             if (Enabled)
+            {
                 WindowManager.SelectedControl = tbPassword;
+                if (!privateMessagingWindow.Enabled) return;
+                pmWindowWasEnabled = true;
+                privateMessagingWindow.Disable();
+            } 
+            else if(pmWindowWasEnabled)
+            {
+                privateMessagingWindow.Enable();
+            }
```

Comment:
The updates to this function cover the scenario where a user is attempting to join another user from within the private messaging window and the game they're attempting to join requires a password. These updates will "hide" the primate messaging window when the password request window opens. It will then re-open the private messaging window when the password request window is canceled. In the scenario where the user specifies a password and clicks OK, the private messaging window is NOT reopened (see the OK btn handler function).